core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/WorkspaceMigrationMonitorSpec.scala

```diff
@@ -153,32 +153,54 @@ class WorkspaceMigrationMonitorSpec
     }
   }
 
-  val stsMock = new GoogleStorageTransferService[IO] {
-    override def getStsServiceAccount(project: GoogleProject): IO[ServiceAccount] =
-      IO.raiseError(new NotImplementedError("getStsServiceAccount is stubbed"))
+  "deleteWorkspaceBucket" should "delete the workspace bucket and record when it was deleted" in {
+    val sourceProject = "general-dev-billing-account"
+    val destBucket = "v1-migration-test-" + UUID.randomUUID.toString.replace("-", "")
 
-    override def createTransferJob(jobName: JobName, jobDescription: String,
-                                   projectToBill: GoogleProject, originBucket: GcsBucketName,
-                                   destinationBucket: GcsBucketName, schedule: JobTransferSchedule): IO[TransferJob] =
-      IO.pure {
-        TransferJob.newBuilder()
-          .setName(s"${jobName}")
-          .setDescription(jobDescription)
-          .setProjectId(s"${projectToBill}")
-          .build
+    val v1Workspace = minimalTestData.v1Workspace.copy(
+      namespace = sourceProject,
+      googleProjectId = GoogleProjectId(sourceProject),
+      bucketName = destBucket
+    )
+
+    withMinimalTestDatabase { _ =>
+      runAndWait {
+        DBIO.seq(
+          workspaceQuery.createOrUpdate(v1Workspace),
+          WorkspaceMigrationMonitor.schedule(v1Workspace)
+        )
       }
 
-    override def getTransferJob(jobName: JobName, project: GoogleProject): IO[TransferJob] =
-      IO.raiseError(new NotImplementedError("getTransferJob is stubbed"))
+      val migration = runAndWait(
+        WorkspaceMigrationMonitor.workspaceMigrations
+          .filter(_.workspaceId === v1Workspace.workspaceIdAsUUID).result
+      )
+        .head
 
-    override def listTransferOperations(jobName: JobName, project: GoogleProject): IO[Seq[Operation]] =
-      IO.raiseError(new NotImplementedError("listTransferOperations is stubbed"))
+      val writeAction = WorkspaceMigrationMonitor.deleteWorkspaceBucket(
+        migration,
+        v1Workspace,
+        new MockGoogleStorageService[IO] {
+          override def deleteBucket(googleProject: GoogleProject, bucketName: GcsBucketName, isRecursive: Boolean, bucketSourceOptions: List[Storage.BucketSourceOption], traceId: Option[TraceId], retryConfig: RetryConfig): fs2.Stream[IO, Boolean] =
```

Comment:
Do you think this implementation is going to lead to extraneous boilerplate if this is reused? Or do you think the clarity is worth the boilerplate? Could we add a default implementation and then have individual tests override it where they need specific implementation?

Reply:
Not 100% either way. I feel it's use here does help with clarity but the duplicate code irks me. That said there's a lot of duplicate code in theses tests and I think it'll be worth tidying up later on (ie not now as we're prototyping)

Reply:
Sure, that sounds fine to me.